src/parsimmon.js

```diff
@@ -210,6 +210,92 @@ function bitSeqObj(namedAlignments) {
   });
 }
 
+function parseBufferFor(other, length) {
+  ensureBuffer();
+  return new Parsimmon(function(input, i) {
+    if (i + length > input.length) {
+      return makeFailure(i, length + " bytes for " + other);
+    }
+    return makeSuccess(i + length, input.slice(i, i + length));
+  });
+}
+
+function parseBuffer(length) {
+  return parseBufferFor("buffer", length).map(function(unsafe) {
+    return Buffer.from(unsafe);
+  });
+}
+
+function encodedString(encoding, length) {
+  return parseBufferFor("string", length).map(function(buff) {
+    return buff.toString(encoding);
+  });
+}
+
+function uintBE(length) {
+  if (length < 0 || length > 6) {
+    throw new Error("uintBE requires length in range [0, 6].");
+  }
```

Comment:
Might be worth checking the number is an integer also. And then putting the logic into a helper function for reuse.

`typeof length === "number" && Math.floor(length) === length` is probably the clearest integer check in JS?